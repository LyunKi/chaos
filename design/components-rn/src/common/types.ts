import { KV, Fn, Prefix } from '@cloud-dragon/common-types';
import { ReactNode } from 'react';
import { StyleProp } from 'react-native';
import { COUNTRIES } from './constants';

export type CloudDesignTheme = KV<any>;

export type ThemeMode = 'light' | 'dark';

export type ThemePack = {
  light: CloudDesignTheme;
  dark: CloudDesignTheme;
};

export type ThemeStyle<ComponentStyle> = {
  [key in keyof ComponentStyle]?: ComponentStyle[key] | Prefix<'$'>;
};

export type Themed<Props, ComponentStyle> = Props & {
  ts?: ThemeStyle<ComponentStyle>;
  style?: StyleProp<ComponentStyle>;
};

export type Testable<Props = any> = Props & {
  testID?: string;
};

export type CloudDesignWrap<Props, ComponentStyle> = Testable<
  Themed<Props, ComponentStyle>
>;

export type RenderProp<T extends any[] = any[]> = Fn<T, ReactNode>;

export type Id = string | number;

export interface AccessoryProps {
  color: string;
  size: number;
}

export type AccessoryRenderProp = (props?: AccessoryProps) => ReactNode;

export interface FormError {
  type: 'warning' | 'error';
  msg: string;
}

export type CountryCode = keyof typeof COUNTRIES;

export type SupportedLocale = 'zh_CN' | 'en_US';

export type Country = (typeof COUNTRIES)['CN'];

export type Size = 'xs' | 'sm' | 'md' | 'lg' | 'xl';

export type Status = 'success' | 'error' | 'warning' | 'info';

export interface MaskProps {
  ts?: KV;
  disableCloseOnPress?: boolean;
}
