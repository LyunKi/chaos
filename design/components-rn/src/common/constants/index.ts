export { default as COUNTRIES } from './countries.json';
export const FONT_BASE = 16;
