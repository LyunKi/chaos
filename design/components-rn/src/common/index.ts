export * from './i18n';
export * from './theme';
export * from './types';
export * from './constants';
export * from './utils';
