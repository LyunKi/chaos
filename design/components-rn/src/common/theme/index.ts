export * from './config';
export * from './theme-manager';
export * from './chakra';
