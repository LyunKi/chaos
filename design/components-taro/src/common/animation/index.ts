import { ViewStyle } from 'react-native';
import { Animation, AnimationConfig } from './animation';
import { ZoomAnimation } from './zoomAnimation';
import { PulseAnimation } from './pulseAnimation';
import { ShakeAnimation } from './shakeAnimation';

export type { AnimationConfig };

export type IconAnimation = Animation<AnimationConfig, ViewStyle>;

export interface IconAnimationRegistry {
  zoom: IconAnimation;
  pulse: IconAnimation;
  shake: IconAnimation;
}

export type SupportedAnimation = keyof IconAnimationRegistry;

export function getIconAnimation(
  animation?: SupportedAnimation,
  config?: AnimationConfig
): IconAnimation | undefined {
  switch (animation) {
    case 'zoom':
      return new ZoomAnimation(config) as IconAnimation;
    case 'pulse':
      return new PulseAnimation(config) as IconAnimation;
    case 'shake':
      return new ShakeAnimation(config) as IconAnimation;
  }
}
