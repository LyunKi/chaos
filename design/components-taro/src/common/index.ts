export * from './i18n';
export * from './theme';
export * from './types';
export * from './constants';
export * from './animation';
export * from './utils';
