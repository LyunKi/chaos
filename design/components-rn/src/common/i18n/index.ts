import { VariableManager, VariablePacks } from '@cloud-dragon/common-utils';
import { SupportedLocale } from '../types';
import { DEFAULT_I18N_KEY } from '../constants';

class I18nManagerClass extends VariableManager {
  public get locale(): SupportedLocale {
    return (this.packKey as any) ?? DEFAULT_I18N_KEY;
  }

  public get i18n() {
    return this.variables;
  }

  public setLocale(locale: SupportedLocale) {
    this.setPackKey(locale);
  }

  public setI18nPacks(i18nPacks: VariablePacks) {
    this.setPacks(i18nPacks);
  }
}

export const I18nManager = new I18nManagerClass({}, DEFAULT_I18N_KEY);
