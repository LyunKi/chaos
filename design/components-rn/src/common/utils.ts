import { merge } from 'lodash';
import { ThemeManager } from './theme';
import { CloudDesignTheme } from './types';

export function extendTheme(themePack: CloudDesignTheme) {
  return merge({}, ThemeManager.getThemePack(), themePack);
}
