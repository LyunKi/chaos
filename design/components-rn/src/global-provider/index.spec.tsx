import React, { useState } from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { GlobalProvider, Text, View } from '../';
import { ThemeMode, ThemeManager } from '../common';

const Case = (props: any) => {
  const [theme, setTheme] = useState<ThemeMode>('light');
  return (
    <GlobalProvider {...props} themeMode={theme}>
      <View ts={{ background: '$color.bg.layout' }}>
        <Text
          testID="text"
          value={ThemeManager.themedValue('$color.font.default')}
        />
      </View>
      <button
        onClick={() => {
          setTheme((prev) => (prev === 'light' ? 'dark' : 'light'));
        }}
      >
        change theme
      </button>
    </GlobalProvider>
  );
};

describe('GlobalProvider', () => {
  test('controls theme', async () => {
    const canvas = render(<Case />);
    const text = await canvas.findByTestId('text');
    const beforeContent = text.textContent;
    const button = await canvas.findByText('change theme');
    await fireEvent.press(button);
    expect(text.textContent !== beforeContent).toBeTruthy();
  });
});
