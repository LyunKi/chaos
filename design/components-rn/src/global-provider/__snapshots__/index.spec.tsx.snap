// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalProvider controls theme 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
/>
`;
